File: backend/models/resort.dart
class Resort {
  final String id;
  final String name;
  final String location;
  final double pricePerNight;
  final String imageUrl;
  final String description;

  Resort({
    required this.id,
    required this.name,
    required this.location,
    required this.pricePerNight,
    required this.imageUrl,
    required this.description,
  });

  factory Resort.fromJson(Map<String, dynamic> json) {
    return Resort(
      id: json['id'] as String,
      name: json['name'] as String,
      location: json['location'] as String,
      pricePerNight: (json['pricePerNight'] as num).toDouble(),
      imageUrl: json['imageUrl'] as String,
      description: json['description'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'location': location,
      'pricePerNight': pricePerNight,
      'imageUrl': imageUrl,
      'description': description,
    };
  }
}

File: backend/models/resort.py
"""Resort Pydantic models."""

from pydantic import BaseModel, Field
from typing import List, Optional

class ResortBase(BaseModel):
    name: str = Field(..., example="Seaside Paradise")
    description: Optional[str] = Field(None, example="A beautiful beach resort.")
    location: str = Field(..., example="Busan, South Korea")
    price_per_night: float = Field(..., example=250.0)
    amenities: List[str] = Field(default_factory=list, example=["Pool", "Spa", "Gym"])

class ResortCreate(ResortBase):
    pass

class ResortOut(ResortBase):
    id: int

    class Config:
        orm_mode = True
